### PR TITLE
Euclid algorithm in bash

### PR DESCRIPTION
**Pull request by Ariana1729**
umm quite useless but slightly better than writing in bf haha, also displayed nicely on gitbook too:>

### CODE REVIEW

File: contents/thomas_algorithm/thomas_algorithm.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Thomas Algorithm
+s# Thomas Algorithm
```
Comment (jiegillet) [inline contents/thomas_algorithm/thomas_algorithm.md]: you typed an `s` here

Comment (jiegillet) [reply contents/thomas_algorithm/thomas_algorithm.md]: Wait this file should not be here at all!

Comment (Ariana1729) [reply contents/thomas_algorithm/thomas_algorithm.md]: oops got to undo commits

File: contents/thomas_algorithm/thomas_algorithm.md [hunk 2]
```diff
@@ -111,8 +111,6 @@ You will find this algorithm implemented [in this project](https://scratch.mit.e
 [import, lang:"java"](code/java/thomas.java)
 {% sample lang="hs" %}
 [import, lang:"haskell"](code/haskell/thomas.hs)
-{% sample lang="go" %}
-[import, lang:"go"](code/golang/thomas.go)
```
Comment (jiegillet) [inline contents/thomas_algorithm/thomas_algorithm.md]: You're still removing lines, you shouldn't touch `thomas_algorithm.md` at all

Comment (Ariana1729) [reply contents/thomas_algorithm/thomas_algorithm.md]: oops yeah

File: book.json [hunk 3]
```diff
@@ -163,6 +163,10 @@
         {
           "lang": "lolcode",
           "name": "LOLCODE"
+        },
+        {
+          "lang": "bash",
```
Comment (leios) [inline book.json]: We usually make `lang` the file extension so `sh` in this case

File: contents/euclidean_algorithm/code/bash/euclid.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
+abs() {
+	local ret=$1
+	if [ $ret -lt 0 ]; then
```
Comment (berquist) [inline contents/euclidean_algorithm/code/bash/euclid.sh]: `[ ... ]` is `sh` (the original Bourne shell). Please use `[[ ... ]]` for consistency.

File: contents/euclidean_algorithm/code/bash/euclid.sh [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+		let "ret = $ret * -1"
```
Comment (berquist) [inline contents/euclidean_algorithm/code/bash/euclid.sh]: While we are working with integers, do general arithmetic instead, using more modern Bash:
```
((ret *= -1))
```

File: contents/euclidean_algorithm/code/bash/euclid.sh (hunk 4 above)
Comment (berquist) [reply contents/euclidean_algorithm/code/bash/euclid.sh]: Though, you can do even better considering the `((...))` syntax (left as an exercise for the reader.

File: contents/euclidean_algorithm/code/bash/euclid.sh [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+euclid_mod() {
+	local a=$(abs $1)
+	local b=$(abs $2)
+	
+	while [ $b -ne 0 ]; do
+		let "tmp = $b"
+		let "b = $a % $b"
+		let "a = tmp"
+	done
+	echo $a
```
Comment (berquist) [inline contents/euclidean_algorithm/code/bash/euclid.sh]: Prefer `printf` for function returns. The `echo`s at the end of the script are fine.

File: contents/euclidean_algorithm/code/bash/euclid.sh [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+		if [ $a -gt $b ]; then
+			let "a -= $b"
+		else
+			let "b -= $a"
+		fi
+	done
+	echo $a
+}
+
+result=$(euclid_mod 143 693)
```
Comment (berquist) [inline contents/euclidean_algorithm/code/bash/euclid.sh]: Use the other examples for consistency:
```bash
result=$(euclid_mod $((64 * 67)) $((64 * 81)))
```
etc.

**Comment by berquist** [comment]
Oh, and
4. rename the extension from `*.sh` to `*.bash`. This is another terrible habit that everyone has (including me).

**Comment by Ariana1729** [comment]
all my bash scripts are also .sh haha, ok editing

**Comment by berquist** [comment]
Not specific to the review, and I'll document it elsewhere, but

1. It is good practice to use [ShellCheck](https://www.shellcheck.net/) on all your shell scripts. Because of their nature, linters/static analysis to help find problems or smells are more important for shell scripts than most programming languages.
2. A good style guide to follow is https://github.com/bahamas10/bash-style-guide.

**Comment by leios** [comment]
Ok. Yeah, you guys are right. The scripts don't need to be `.sh`. Sorry for that.

**Comment by Ariana1729** [comment]
oh wow shellcheck seems quite cool, giong to update code based on that

**Comment by berquist** [comment]
The most common shellcheck error is about quoted variables. While this is _the_ most important thing for daily scripting, it isn't important here at all, so ignore it.

**Comment by berquist** [comment]
You still need to fix the filename.

**Comment by Ariana1729** [comment]
Ok should be fixed

**Comment by berquist** [comment]
I can't pick it apart further, thanks!

**Pull request merged**